### DISCUSSION TITLE
Broken Business rules list

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#3222] [Questions]
Almost all of our business objects have business rules ranging from 'Required' to something relatively complex. Instead of showing users dozens of broken business rules, we are thinking if we could show users a subset e.g. list of broken business rules with top priority and when they are 'met' then show users next level of broken business rules (based on priority).
We do specify priority for our business rules.

Kind regards

**Comment by rockfordlhotka** [comment]
There's an API you can use to get a list of the _rules_ for an object. I don't know if priority is in that info, but it probably could be. You could use LINQ to join the broken rules with the rules (I hope).

**Reply by Chicagoan2016** [reply]
@rockfordlhotka Thank you.
I would like to make a couple of suggestions for the next version of Csla
1) We could add 'GetFirstBrokenRules()' in BrokenRulesCollection, this method will return 'first' broken rules for the object (or object graph).
2) Add Priority to BrokenRule.

Kind Regards

**Reply by rockfordlhotka** [reply]
It has been too many years, and I don't recall why the broken rule objects don't maintain a reference to the rule object that was broken, which seems like it would be a good solution. 

I don't know about `GetFirstBrokenRules`, because people's definition of "first" might vary. It seems better (to me) to provide more info and let people use a LINQ query to order the broken rules in a way that meets the app's specific requirements.

**Reply by Chicagoan2016** [reply]
@rockfordlhotka  I was wondering how is 'first' determined in GetFirstBrokenRule(..) in BrokenRulesCollection?

**Reply by rockfordlhotka** [reply]
iirc, it is the [0] element in the array - so literally the first rule that was broken in chronological order.
